logfire/_internal/integrations/llm_providers/openai.py

```diff
@@ -1,16 +1,17 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, cast
+from typing import TYPE_CHECKING, Any, cast
 
 import openai
 from openai._legacy_response import LegacyAPIResponse
+from openai.lib.streaming.chat._completions import ChatCompletionStreamState
```

Comment:
I'm concerned about this private import. Do we have other options? Can you check if this works for older openai versions?

@Kludex shouldn't this have been flagged by pre-commit?

Reply:
Which hook? This is already in `logfire._internal`.

Reply:
https://github.com/microsoft/pyright/issues/9008

😅

Reply:
This was added in https://github.com/openai/openai-python/releases/tag/v1.40.0 released in August. So definitely makes sense to add handling for versions of the `openai` package that do not have it. What would be the best way to do that? / Is there an example of something similar to this in logfire already? My initial thought is to try/except the import and fall back to using the `OpenaiCompletionStreamState` if this is not available.

In terms of other options, there isn't a public function for this in the openai package. Similar code for recombining stream chunks into a message object could be added in logfire. It does seem like a broadly useful capability, so I will open an issue for it with openai.

Reply:
openai issue: https://github.com/openai/openai-python/issues/1833

Reply:
> My initial thought is to try/except the import and fall back to using the `OpenaiCompletionStreamState` if this is not available.

That sounds right. And thanks for opening an issue.

Reply:
Will you add a try/except to the import?

Reply:
Added now in b5b26e8752514fab527b1487717c1d3d57661ad9. I just try-excepted the whole class definition, which is similar to how `_ProxyGauge` dependency on `Gauge` is done here https://github.com/jackmpcollins/logfire/blob/b5b26e8752514fab527b1487717c1d3d57661ad9/logfire/_internal/metrics.py#L309-L322  Please let me know if you have a preferred way to do this.